datafusion/expr/src/udf.rs

```diff
@@ -337,6 +353,11 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
     fn monotonicity(&self) -> Result<Option<FuncMonotonicity>> {
         Ok(None)
     }
+
+    // Do the function rewrite
+    fn simplify(&self, _args: &[Expr]) -> Result<Simplified> {
+        Ok(Simplified::Original)
+    }
```

Comment:
Is it possible to make this take `self` (and avoid a copy)? Something like

```suggestion
    fn simplify(self, _args: &[Expr]) -> Result<Simplified> {
        Ok(Simplified::Original(self))
    }
```?

Reply:
It seems quite hard.

Reply:
![Screenshot 2024-02-28 at 08 43 25](https://github.com/apache/arrow-datafusion/assets/15869383/ec308c4c-a72f-4aef-8930-f4b7cad91c5b)

Reply:
If I +Sized for UDF trait, it can't be made into an object.